client/subscription-edit-page/index.tsx

```diff
@@ -0,0 +1,285 @@
+		};
+	}
+
+	// Wrap in an internal callback to load the select's value and tokens.
+	const internalCallback = async () => {
+		const newUserId = parseInt( customerUserSelect.value, 10 ) || 0;
+
+		setUser( newUserId );
+
+		// Loaded, loading, or errored out, we do not need to load anything.
+		if ( hasEntry( cache, newUserId ) ) {
+			return;
+		}
+
+		const cacheWithLoading = startLoading( cache, newUserId );
+		setCache( cacheWithLoading );
+
+		try {
+			const response = await fetchUserTokens( newUserId, ajaxUrl, nonce );
+			if ( undefined === response ) {
+				throw new Error(
+					__(
+						'Failed to fetch user tokens. Please reload the page and try again.',
+						'woocommerce-payments'
+					)
+				);
+			}
+
+			setCache(
+				tokensLoaded( cacheWithLoading, newUserId, response.tokens )
+			);
+		} catch ( error ) {
+			setCache(
+				loadingFailed(
+					cacheWithLoading,
+					newUserId,
+					error instanceof Error
+						? error.message
+						: __( 'Unknown error', 'woocommerce-payments' )
+				)
+			);
+		}
+	};
+
+	// Add the listener with the right technique, as select2 does not emit <select> events.
+	jQuery( customerUserSelect ).on( 'select2:select', internalCallback );
+	customerUserSelect.addEventListener( 'change', internalCallback );
+
+	return () => {
+		jQuery( customerUserSelect ).off( 'select2:select', internalCallback );
+		customerUserSelect.removeEventListener( 'change', internalCallback );
+	};
+};
+
+/**
+ * Renders a payment method select or loading indicator.
+ */
+export const PaymentMethodSelect = ( {
+	inputName,
+	initialValue,
+	initialUserId,
+	initialCache,
+	nonce,
+	ajaxUrl,
+}: PaymentMethodSelectProps ) => {
+	const [ value, setValue ] = useState< number >( initialValue );
+	const [ userId, setUserId ] = useState< number >( initialUserId );
+	const [ cache, setCache ] = useState< CachedUserData >( initialCache );
+
+	// If there is no value but a user, try to fall back to the default token.
+	useEffect( () => {
+		if ( 0 !== userId && value === 0 ) {
+			const defaultTokenId = getDefaultTokenId( cache, userId );
+			if ( value !== defaultTokenId ) {
+				setValue( value );
+			}
+		}
+	}, [ userId, value, cache ] );
+
+	// Use the listener to update the user ID and cache.
+	useEffect( () => {
+		return addCustomerSelectListener(
+			cache,
+			ajaxUrl,
+			nonce,
+			setUserId,
+			setCache
+		);
+	}, [ userId, cache, ajaxUrl, nonce ] );
+
+	if ( userId <= 0 ) {
+		return (
+			<select name={ inputName } defaultValue={ 0 } key={ 'no-customer' }>
+				<option value={ 0 } key={ 'no-customer' } disabled>
+					{ __(
+						'Please select a customer first',
+						'woocommerce-payments'
+					) }
+				</option>
+			</select>
+		);
+	}
+
+	// There might be a user, but tokens could still be loading.
+	const entry = getUserEntry( cache, userId );
+	if ( undefined === entry || entry.loading ) {
+		return <>{ __( 'Loading…', 'woocommerce-payments' ) }</>;
+	} else if ( entry.loadingError ) {
```

Comment:
```suggestion
	}
	if ( entry.loadingError ) {
```

( ⛏️ ) since there's an early return on the block above, the `else` doesn't seem to be needed

Reply:
Separated in https://github.com/Automattic/woocommerce-payments/pull/11143/commits/7257a80c0d4beac8a5e1419e35663f6e56d4539a.